test/lib/router.js

```diff
@@ -1260,6 +1260,35 @@ describe('Router', function() {
         done();
       });
     });
+
+    it('places a `_matchedRoute` value on context', function(done) {
+      var app = koa();
+      var router = new Router();
+      var middleware = function *(next) {
+        expect(this._matchedRoute).to.be('/users/:id')
+        yield next;
+      };
+
+      router.use(middleware);
+      router.get('/users/:id', function *() {
+        expect(this._matchedRoute).to.be('/users/:id')
+        should.exist(this.params.id);
+        this.body = { hello: 'world' };
+      });
+
+      var routerMiddleware = router.routes();
```

Comment:
if you add another similar but more generic route here, the test will fail for the above route
```
router.get('/:username/:id', function *() {
    // this._matchedRoute for '/users/:id' will now instead be '/:username/:id'
})
```